src/rebar_templater.erl

```diff
@@ -266,17 +266,27 @@ replace_var([H|T], Acc, Vars) ->
 
 %% Load a list of all the files in the escript and on disk
 find_templates(State) ->
-    %% Cache the files since we'll potentially need to walk it several times
-    %% over the course of a run.
-    Files = cache_escript_files(State),
-
-    %% Build a list of available templates
-    AvailTemplates = prioritize_templates(
-        tag_names(find_disk_templates(State)),
-        tag_names(find_escript_templates(Files))),
-
-    ?DEBUG("Available templates: ~p\n", [AvailTemplates]),
-    {AvailTemplates, Files}.
+    case rebar_state:escript_path(State) of
+        undefined ->
+            AvailTemplates = prioritize_templates(
+                               tag_names(find_disk_templates(State)),
+                               tag_names(find_priv_templates(State))),
+
+            ?DEBUG("Available templates: ~p\n", [AvailTemplates]),
+            {AvailTemplates, []};
+        _ ->
+            %% Cache the files since we'll potentially need to walk it several times
+            %% over the course of a run.
+            Files = cache_escript_files(State),
+
+            %% Build a list of available templates
+            AvailTemplates = prioritize_templates(
+                               tag_names(find_disk_templates(State)),
+                               tag_names(find_escript_templates(Files))),
+
+            ?DEBUG("Available templates: ~p\n", [AvailTemplates]),
+            {AvailTemplates, Files}
+    end.
```

Comment:
Can you just break these two into like `find_priv_templates(State)` and `find_escript_templates(State)` for the search in order to reduce some level of code duplication?